### PR TITLE
Always keep Spacy PoS tagger on

### LINKED ISSUE #5036: spacy 3.0 warnings about lemmatization and POS
I'm training a model (https://github.com/allenai/allennlp-models/blob/main/training_config/pair_classification/mnli_roberta.jsonnet) with allennlp 2.1.0, using SpaCy 3.

There are a bunch of warnings that show up (I'm not sure if these were here before, but I noticed them now because my log files are now massive):

```
[W108] The rule-based lemmatizer did not find POS annotation for the token 'Tommy'. Check that your pipeline includes components that assign token.pos, typically 'tagger'+'attribute_ruler' or 'morphologizer'.
[W108] The rule-based lemmatizer did not find POS annotation for the token 'hesitated'. Check that your pipeline includes components that assign token.pos, typically 'tagger'+'attribute_ruler' or 'morphologizer'.
[W108] The rule-based lemmatizer did not find POS annotation for the token '.'. Check that your pipeline includes components that assign token.pos, typically 'tagger'+'attribute_ruler' or 'morphologizer'.
[W108] The rule-based lemmatizer did not find POS annotation for the token 'Tommy'. Check that your pipeline includes components that assign token.pos, typically 'tagger'+'attribute_ruler' or 'morphologizer'.
[W108] The rule-based lemmatizer did not find POS annotation for the token 'hesitated'. Check that your pipeline includes components that assign token.pos, typically 'tagger'+'attribute_ruler' or 'morphologizer'.
[W108] The rule-based lemmatizer did not find POS annotation for the token 'for'. Check that your pipeline includes components that assign token.pos, typically 'tagger'+'attribute_ruler' or 'morphologizer'.
[W108] The rule-based lemmatizer did not find POS annotation for the token 'a'. Check that your pipeline includes components that assign token.pos, typically 'tagger'+'attribute_ruler' or 'morphologizer'.
[W108] The rule-based lemmatizer did not find POS annotation for the token 'short'. Check that your pipeline includes components that assign token.pos, typically 'tagger'+'attribute_ruler' or 'morphologizer'.
...
```

I think this is because `allennlp.common.util.get_spacy_model` (https://github.com/allenai/allennlp/blob/96415b2bab6d8c70a0fa80ca4a8b9d1dc988720e/allennlp/common/util.py#L258) has POS off by default, but doesn't disable lemmatization by default.

Not sure what y'all think is the best way to solve this...can add a lemmatization argument for get_spacy_model that is default by false? This is a change in the defaults from previous versions, though.

### PR DESCRIPTION
**Pull request by leo-liuzy**
Fixes #5036  .

Changes proposed in this pull request:

- We always keep our spacy model's PoS tagger on, since the additional computation seems small and fix the behavioral issue of upgrading to SpaCy 3.0
-

### CODE REVIEW

File: allennlp/common/util.py [hunk 1]
```diff
@@ -256,7 +256,7 @@ def prepare_environment(params: Params):
 
 
 def get_spacy_model(
-    spacy_model_name: str, pos_tags: bool, parse: bool, ner: bool
+    spacy_model_name: str, parse: bool, ner: bool, pos_tags: bool = True
```
Comment (epwalsh) [inline allennlp/common/util.py:259]: I don't like the idea of changing to order of arguments if it's not necessary. Can we just do this?

```suggestion
    spacy_model_name: str, pos_tags: bool = True, parse: bool = False, ner: bool = False
```

Comment (leo-liuzy) [reply allennlp/common/util.py:259]: Make sense :-)

File: CHANGELOG.md [hunk 2]
```diff
@@ -28,6 +28,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Makes sure tensors that are stored in `TensorCache` always live on CPUs
 - Fixed a bug where `FromParams` objects wrapped in `Lazy()` couldn't be pickled.
 - Fixed a bug where the `ROUGE` metric couldn't be picked.
+- Fixed a bug reported by https://github.com/allenai/allennlp/issues/5036 . We keeps our spacy POS tagger on.
```
Comment (epwalsh) [inline CHANGELOG.md:31]:
```suggestion
- Fixed a bug reported by https://github.com/allenai/allennlp/issues/5036. We keeps our spacy POS tagger on.
```

**Review by epwalsh — APPROVED** [review]
LGTM! Thanks @leo-liuzy!

**Pull request merged**